tcptracer-bpf.c

```diff
@@ -864,17 +853,94 @@ int kretprobe__inet_csk_accept(struct pt_regs *ctx)
 		return 0;
 	}
 
+    struct fd_info t = { .active = 1 };
+    t.start_time_ns = bpf_ktime_get_ns();
+
+    bpf_map_update_elem(&active_fds, &newsk, &t, BPF_ANY);
+
 	return assert_tcp_record(newsk, status, DIRECTION_INCOMING, STATE_INITIALIZING);
 }
 
+// Is kretprobe__inet_csk_accept sufficient or is this one needed as well or instead?
+
+// SEC("kprobe/__x64_sys_accept4")
+// int kprobe__x64_sys_accept4(struct pt_regs *ctx)
+// {
+	// The file descriptor is the value returned from the syscall.
+	// int fd = PT_REGS_RC(ctx);
+	// bool t = true;
+	// bpf_map_update_elem(&active_fds, &fd, &t, BPF_ANY);
+
+// 	return 0;
+// }
+
+SEC("kprobe/__x64_sys_writev")
+int kprobe__sys_writev(struct pt_regs *ctx)
```

Comment:
Is this probe needed? It seems like the `tcp_sendmsg` probe does all the actual work. If it is required it needs some comments to explain this to our future selfs.

Reply:
gone